### ISSUE TITLE
Add support for metadata information

### ISSUE THREAD

**Issue by liaoyuan1011** [CycloneDX/cyclonedx-ruby-gem#19]
https://cyclonedx.org/docs/1.4/json/#metadata

**Comment by stevespringett** [comment]
Yes, there are, but we rely primarily on community contributions. Support for v1.5 (to be released soon) is certainly in scope.

**Comment by stevespringett** [comment]
Ideally the following metadata should be supported:
- metadata.timestamp
- metadata.lifecycles.phase (requires CycloneDX v1.5 support)
- metadata.tools
- metadata.component
